icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/RbnfTest.java

```diff
@@ -1162,6 +1191,26 @@ public void TestAllLocales() {
                         logln(msg);
                         errors.append("\n" + msg);
                     }
+
+                    try {
+                        // RBNF parse is extremely slow when lenient option is enabled.
+                        fmt.setLenientParseMode(true);
+                        Number num = fmt.parse(s);
+                        if (isVerbose()) {
+                            logln(loc.getName() + names[j] + "success parse: " + s + " -> " + num);
+                        }
+                        if (j != 0) {
+                            // TODO: Fix the ordinal rules.
+                            continue;
+                        }
+                        if (n != num.doubleValue()) {
+                            errors.append("\n" + loc + names[j] + "got " + num + " expected " + n);
+                        }
+                    } catch (ParseException pe) {
```

Comment:
This code seems redundant with the previous block. Please try to use a do while loop that just enumerates a lenient mode boolean from false to true?